### PR TITLE
Rename the dependency table to package

### PR DESCRIPTION
**Pull request by mprahl**
This will allow a future change to reuse this table to map the top level packages associated with the request.

Relates to FACTORY-5064

### CODE REVIEW

**Comment by mprahl** [comment]
@lcarva I refactored it so that the API is better. Could you please take a look again?

**Comment by yashvardhannanavati** [comment]
:+1:

File: cachito/web/models.py [hunk 1]
```diff
@@ -67,8 +67,9 @@ def get_state_names(cls):
         return sorted([state.name for state in cls])
 
 
-class Dependency(db.Model):
-    """A dependency (e.g. gomod dependency) associated with the request."""
+class Package(db.Model):
+    """A package associated with the request."""
+    __tablename__ = 'package'
```
Comment (lcarva) [inline cachito/web/models.py]: Can we add a comment so we don't accidentally remove this in the future?

File: cachito/web/models.py [hunk 2]
```diff
@@ -77,10 +78,19 @@ class Dependency(db.Model):
         db.UniqueConstraint('name', 'type', 'version'),
     )
 
+    def __repr__(self):
+        return f'<Package id={self.id}, name={self.name} type={self.type} version={self.version}>'
```
Comment (lcarva) [inline cachito/web/models.py]: Can we just use `self.__class__.__name__` so we don't have to override it on every sub class?

**Pull request merged**